packages/manager/modules/pci/src/projects/project/project.service.js

```diff
@@ -20,9 +20,45 @@ export default class {
       .catch(() => ({}));
   }
 
-  getCustomerRegions(serviceName) {
+  getCustomerRegions(serviceName, withPaginationMode = false) {
+    const requestHeader = withPaginationMode
+      ? {
+          'X-Pagination-Mode': 'CachedObjectList-Pages',
+        }
+      : null;
     return this.$http
-      .get(`/cloud/project/${serviceName}/region`)
+      .get(`/cloud/project/${serviceName}/region`, {
+        headers: requestHeader,
+      })
       .then(({ data: regions }) => regions);
   }
+
+  getStorageRegions(projectId, regionCapacity) {
+    return this.getCustomerRegions(projectId, true).then((regions) => {
+      return regions.filter(({ services }) =>
+        services.find(({ name }) => name === regionCapacity),
+      );
+    });
+  }
+
+  getS3StorageRegions(projectId, regionCapacity) {
+    return this.getCustomerRegions(projectId, true).then((regions) => {
+      return regions.filter(({ services }) =>
+        services.find(
+          ({ name }) =>
+            name === regionCapacity[0] || name === regionCapacity[1],
+        ),
+      );
+    });
+  }
+
+  getRegions(projectId) {
+    return this.$http
+      .get(`/cloud/project/${projectId}/region`, {
+        headers: {
+          'X-Pagination-Mode': 'CachedObjectList-Pages',
+        },
+      })
+      .then(({ data }) => data);
```

Comment:
**question:** Since the function `getCustomerRegions` has been updated, can we remove this function?

Reply:
Done Anoop